src/twisted/internet/defer.py

```diff
@@ -2262,12 +2283,16 @@ def run(
         @return: L{Deferred} of function result.
         """
 
-        def execute(ignoredResult: object) -> Deferred[_SelfResultT]:
-            return maybeDeferred(f, *args, **kwargs).addBoth(self._releaseAndReturn)
+        def execute(ignoredResult: object) -> Deferred[_T]:
+            # maybeDeferred arg type requires one of the possible union members
+            # and won't accept all possible union members
+            return maybeDeferred(f, *args, **kwargs).addBoth(
+                self._releaseAndReturn
+            )  # type: ignore[return-value]
```

Comment:
I tried to get this to pass type checking but I can't work out how: https://mypy-play.net/?mypy=latest&python=3.12&gist=7a0b28937ca2f2137a55f29610da1d16&flags=strict